javascript/main.js

```diff
@@ -0,0 +1,119 @@
+const form = document.getElementById('sign-up-form');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const formData = new FormData(e.target);
+  const errors = validateForm(formData);
+
+  document.getElementById('email-error').innerText = errors.email;
+  document.getElementById('username-error').innerText = errors.username;
+  document.getElementById('password-error').innerText = errors.password;
+  document.getElementById('confirm-password-error').innerText =
+    errors.confirmPassword;
+
+  if (
+    errors.email ||
+    errors.username ||
+    errors.password ||
+    errors.confirmPassword
+  )
+    return;
+
+  // Loading state for 2 secs & success
+  form.innerHTML = `
+    <div>Loading...</div>
+  `;
+
+  setTimeout(() => {
+    form.innerHTML = `
+      <div>You're signed in ${formData.get('username')}!</div>
+    `;
+  }, 2000);
+});
+
+const validateForm = (formData) => {
+  const errors = {
+    email: '',
+    username: '',
+    password: '',
+    confirmPassword: '',
+  };
+
+  const fieldValues = {};
+
+  for (let [name, value] of formData) {
+    fieldValues[name] = value;
+  }
```

Comment:
```suggestion
const fieldValues = Object.fromEntries(formData);

```

Reply:
I completely forgot about that method 😅. Thanks for the suggestion!